src/home/homework-luks.c

```diff
@@ -2356,8 +2361,13 @@ static int ext4_offline_resize_fs(HomeSetup *setup, uint64_t new_size, bool disc
                 return r;
         if (r == 0) {
                 /* Child */
+                int saved_errno;
+
                 execlp("e2fsck" ,"e2fsck", "-fp", setup->dm_node, NULL);
-                log_error_errno(errno, "Failed to execute e2fsck: %m");
+
+                saved_errno = errno;
+                log_open();
```

Comment:
Is it necessary? `safe_fork()` here does not have `FORK_CLOSE_ALL_FDS`.

Reply:
You're right; I must have missed that somehow. But now that you bring it up, perhaps it _should_ use `FORK_CLOSE_ALL_FDS`?

Reply:
Yeah, please add the flag. Thanks.

Reply:
Added

Reply:
Hmm, I noticed right now that here `FORK_STDOUT_TO_STDERR` is set. I guess it conflicts with `FORK_CLOSE_ALL_FDS`... @poettering could you comment on this?

Reply:
They doesn't seem to conflict; `FORK_CLOSE_ALL_FDS` calls through to `close_all_fds(NULL, 0)`, which closes fds starting from 3.